homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
+SELECT employee_id, first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT MAX(salary)
+    FROM employees
+    WHERE department_id = 40
+);
+
 --41. the department name and Id for all departments where they located, that Id is equal to the Id for the location where department number 30 is located.
+SELECT department_id, department_name
+FROM departments
+WHERE location_id = (
+    SELECT location_id
+    FROM departments
+    WHERE department_id = 30
+);
+
 --42. the first and last name, salary, and department ID for all those employees who work in that department where the employee works who hold the ID 201.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE department_id = (
+    SELECT department_id
+    FROM employees
+    WHERE employee_id = 201
+);
+
 --43. the first and last name, salary, and department ID for those employees whose salary is equal to the salary of the employee who works in that department which ID is 40.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary IN (
+    SELECT salary
+    FROM employees
+    WHERE department_id = 40
+);
+
 --44. the first and last name, salary, and department ID for those employees who earn more than the minimum salary of a department which ID is 40.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE department_id = 40
+);
+
 --45. the first and last name, salary, and department ID for those employees who earn less than the minimum salary of a department which ID is 70.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE department_id = 40
+);
+
 --46. the first and last name, salary, and department ID for those employees who earn less than the average salary, and also work at the department where the employee Laura is working as a first name holder.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary < (
+    SELECT AVG(salary)
+    FROM employees
+)
+  AND department_id = (
+    SELECT department_id
+    FROM employees
+    WHERE first_name = 'Laura'
+);
+
 --47. the full name (first and last name) of manager who is supervising 4 or more employees.
+SELECT (first_name || ' ' || last_name) AS employee_name
+FROM employees
+WHERE employee_id IN (
+    SELECT manager_id
+    FROM employees
+    GROUP BY manager_id
+    HAVING COUNT(manager_id) >= 4
+);
+
 --48. the details of the current job for those employees who worked as a Sales Representative in the past.
+SELECT *
+FROM jobs
+WHERE job_id IN (
+    SELECT job_id
+    FROM employees
+    WHERE employee_id IN(
+        SELECT DISTINCT employee_id
+        FROM job_history
+        WHERE job_id = (
+            SELECT job_id
+            FROM jobs
+            WHERE job_title = 'Sales Representative'
+        )
+    )
+);
```

Comment:
Try always to use at most 2 SUBQUERIES, and even though it will be better to have one, but as you saw there are some cases when you should check 2 conditions from different tables. Here I would see such a query:
What I see here:
1. Use a join to get access to employees (JOIN jobs WITH employees).
2. Check if employee_id is IN the sub-query list.
3. The sub-query will extract employee_id from job_history JOIN jobs based on job_id, and use WHERE for job_title check.